### ISSUE TITLE
rockchip patches not correct

### ISSUE THREAD

**Issue by laroche** [raspberrypi/linux#5071]
### Describe the bug

Looking at the 5.18 branch of this repository. The checkin is d9d409987bf466c52599eda0254c0c3a8d950b1f.

This contains e.g. changes to drivers/media/platform/rockchip/rkisp1/rkisp1-isp.c.

Many definitions to "state" are duplicated in the patch and thus don't even compile.

Correct patch should look like:
<pre>
--- a/drivers/media/platform/rockchip/rkisp1/rkisp1-isp.c
+++ b/drivers/media/platform/rockchip/rkisp1/rkisp1-isp.c
@@ -214,6 +214,9 @@ rkisp1_isp_get_pad_fmt(struct rkisp1_isp *isp,
        struct v4l2_subdev_state state = {
                .pads = isp->pad_cfg
                };
+       struct v4l2_subdev_state state = {
+               .pads = &state
+               };
        if (which == V4L2_SUBDEV_FORMAT_TRY)
                return v4l2_subdev_get_try_format(&isp->sd, sd_state, pad);
        else
</pre>

Of course this is not your preferred target to compile for, maybe you
want to correct this anyway.

best regards,

Florian La Roche

### Steps to reproduce the behaviour

Checkout rpi-5.18 branch.

### Device (s)

Raspberry Pi 3 Mod. B+

### System

Current 5.18 branch in github source tree.

**Comment by 6by9** [comment]
It's also wrong in e375dd2a586273a47bb055c1cf2eba642fcd1df0 on the rpi-5.15.y branch, which also appears to have sucked in a fix for bcm2835-isp into the one patch.

The Rockchip changes look totally erroneous as the driver already has the ```struct v4l2_subdev_state state``` defined via https://github.com/torvalds/linux/commit/0d346d2a6f54f06f36b224fd27cd6eafe8c83be9 in mainline.

**Comment by laroche** [comment]
Right, the patch can be left out completely (for both rockchip files) and not just reduced as written by me above.
A compile with this removed has finished for me.

best regards,

Florian La Roche

**Comment by laroche** [comment]
The 5.19 branch also has this wrong patch included.
